File: app/src/main/java/info/nukoneko/cuc/android/kidspos/ui/calculator/CalculatorView.java
package info.nukoneko.cuc.android.kidspos.ui.calculator;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.widget.AppCompatButton;
import android.util.AttributeSet;
import android.view.Gravity;

import info.nukoneko.cuc.android.kidspos.R;


public final class CalculatorView extends AppCompatButton {
    private final int mNumber;

    public CalculatorView(Context context) {
        this(context, null, 0);
    }

    public CalculatorView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CalculatorView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        final TypedArray typedArray =
                context.obtainStyledAttributes(attrs, R.styleable.CalculatorView, defStyleAttr, defStyleAttr);
        mNumber = typedArray.getInteger(R.styleable.CalculatorView_number, -1);
        typedArray.recycle();

        setGravity(Gravity.CENTER);
        setText(String.valueOf(mNumber));
    }

    public int getNumber() {
        return mNumber;
    }
}

File: app/src/main/kotlin/info/nukoneko/cuc/android/kidspos/ui/main/calculate/CalculatorView.kt
package info.nukoneko.cuc.android.kidspos.ui.main.calculate

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import androidx.appcompat.widget.AppCompatButton
import info.nukoneko.cuc.android.kidspos.R

class CalculatorView : AppCompatButton {
    constructor(context: Context) : this(context, null, 0)
    constructor(context: Context, attr: AttributeSet? = null) : this(context, attr, 0)
    constructor(
        context: Context,
        attr: AttributeSet? = null,
        defStyleAttr: Int = 0
    ) : super(context, attr, defStyleAttr) {
        val typedArray = context.obtainStyledAttributes(
            attr,
            R.styleable.CalculatorView,
            defStyleAttr,
            defStyleAttr
        )
        number = typedArray.getInteger(R.styleable.CalculatorView_number, -1)
        typedArray.recycle()
        gravity = Gravity.CENTER
        text = number.toString()
    }

    val number: Int
}
